source/toolbox/memory/Array.h

```diff
@@ -43,6 +44,76 @@ template <class TYPE> class Pointer;
 template <class TYPE>
 class Array
 {
+private:
+
+   class Allocator
+   {
+   private:
+      constexpr static unsigned int uilog2(unsigned int val) {
+          if (val == 0) return std::numeric_limits<unsigned int>::max();
+          if (val == 1) return 0;
+          unsigned int ret = 0;
+          while (val > 1) {
+              val >>= 1;
+              ret++;
+          }
+          return ret;
+      }
```

Comment:
We can use `std::ilogb()` instead of this function. We should also guarantee that length-0 allocations work correctly.

Reply:
It gained integer overloads in C++11.